File: ebean-autotune/src/test/java/org/tests/model/basic/Customer.java
package org.tests.model.basic;

import io.ebean.annotation.NotNull;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "o_customer")
public class Customer extends BaseModel {

  @NotNull String name;

  String note;

  @ManyToOne(cascade = CascadeType.PERSIST)
  Address billingAddress;

  public Customer(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getNote() {
    return note;
  }

  public void setNote(String note) {
    this.note = note;
  }

  public Address getBillingAddress() {
    return billingAddress;
  }

  public void setBillingAddress(Address billingAddress) {
    this.billingAddress = billingAddress;
  }
}

File: tests/test-kotlin/src/test/kotlin/org/example/order/Customer.kt
package org.example.order

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Version
import org.example.api.ICustomer

@Entity
class Customer(
  name: String,
) : ICustomer {
  @Id
  var id: Int = 0
  val name: String = name

  @Version
  var version: Long = 0
}
